### DISCUSSION TITLE
Override LSP go to definition?

### DISCUSSION THREAD

**Discussion by AlexanderHott** [a-h/templ#918] [Q&A]
Is it possible to override the default behavior of go to definition to navigate to the templ file instead of the generated go file?

I use both neovim and JetBrains Goland, but neither of them have this feature available (that I could find).

I looked a bit into the lsp goto definition spec, and it seems not too difficult to implement. Am I mistaken, or is there a workflow that I'm missing?

**Comment by joerdav** [comment]
Hey, we do have an ongoing discussion around this. The difficulty is that the templ lsp doesn't run on go files: https://github.com/a-h/templ/issues/387

But we do want to solve it, it's just a bit more nuanced than we first thought.
